nautobot/extras/models/change_logging.py

```diff
@@ -213,6 +213,13 @@ def get_snapshots(self):
 
         prior_change = self.get_prev_change(only=["object_data_v2", "object_data"])
 
+        # Deal with the cases where we are trying to capture an object deletion and there is no prior change record.
+        # This can happen when the object is first created and the changelog for that object creation action is deleted.
+        if self.action == ObjectChangeActionChoices.ACTION_DELETE and prior_change is None:
+            prechange = self.object_data_v2
+            if prechange is None:
+                prechange = self.object_data
+
         if self.action != ObjectChangeActionChoices.ACTION_CREATE and prior_change is not None:
```

Comment:
Feels like this should become an `elif` (and the case at line 221/228 below as well) just for clarity that these are mutually exclusive cases?

Reply:
Yeah I got very confused reading the code, will refactor it to something more readable and add comments